### ISSUE TITLE
[Feature] Testing for Pixel 3a XL

### ISSUE THREAD

**Issue by ncorder** [openandroidinstaller-dev/openandroidinstaller#38]
I recently purchased a Pixel 3a XL and am trying to migrate from Ubuntu Touch to Lineage OS.

As long as it isn't damaging to the device, I would love to be a tester for providing support for the machine if possible. 
It is reasonable for me to test the installer with the 3a image? What should I keep in mind beforehand.

**Comment by ncorder** [comment]
Looking through this: https://github.com/openandroidinstaller-dev/openandroidinstaller#how-to-contribute-your-own-installation-configurations

I believe the only thing I must change from the 3a and 3a XL is that the 3a XL's codename is `bonito`

https://github.com/openandroidinstaller-dev/openandroidinstaller/blob/main/openandroidinstaller/assets/configs/sargo.yaml

**Comment by ncorder** [comment]
Does the installer require that I flash Android 12.0.1 first?
https://flash.android.com/build/8561491?target=bonito-user&signed

12.0.1 has a few versions, I can try and test with `SP2A.220505.006`

**Comment by ncorder** [comment]
[openandroidinstaller.log](https://github.com/openandroidinstaller-dev/openandroidinstaller/files/10026654/openandroidinstaller.log)

Appears to have a few error messages through the install.

**Comment by ncorder** [comment]
Pastebin of log: https://paste.debian.net/1260952/

[Screenshot at 2022-11-16 17-30-24](https://user-images.githubusercontent.com/53316485/202324832-34dc5eec-f85e-4586-bf00-135ac7ce6ee7.png)

**Comment by ncorder** [comment]
It appeared to be idling after that step, but to clarify is there anything I must  do on the device after booting into TWRP or should the my laptop continue with the installation after that step?

**Comment by tsterbak** [comment]
Hey, thank you for your interest in the project! :)
I'm looking into the issue and trying to figure out what went wrong here.

This issue might be related to protected/encrypted partitions on the phone that TWRP cannot wipe and so a factory reset is not possible. I will try to reproduce it with a Pixel 3a which should be fairly similar.
Did you enable a device pin/fingerprint or similar on the install of the previous ROM? If yes, you could try to disable all of that and retry installing with the OpenAndroidInstaller.

Thanks a lot for your time and effort!
If you like to contribute your config file, feel free to open a pull request :)

**Comment by tsterbak** [comment]
You are right, currently Android 12.0.1 is required. I'm working on automated checks for making sure this is the case before letting people progress.
Version SP2A.220505.006 should be fine.

**Comment by tsterbak** [comment]
Hey @ncorder, I could partially reproduce the issue with my Pixel 3a and adding a PIN. So removing the PIN and/or Fingerprintlock should fix that issue. Does it work for you?

I will add that to the requirements.

**Comment by ncorder** [comment]
You might edit the screen for selecting "restart bootloader" to include directions for the user to use the
- volume keys to switch between options
- power button to select
Especially if this projects goal is to make it easy for _any non-technical person_ to install a different Android ROM.

**Comment by ncorder** [comment]
![Screenshot at 2022-11-22 21-51-22](https://user-images.githubusercontent.com/53316485/203471891-d7681946-b11b-43b3-80b5-5b8a5316ac65.png)

Log file: https://paste.debian.net/1261563/

**Comment by ncorder** [comment]
I've been selecting "Bootloader is already unlocked" because I have installed LineageOS and Ubuntu Touch on this device before.

Should I start leaving that button unselected? The bootloader should be unlocked.

**Comment by tsterbak** [comment]
Thanks for testing again!

I can see, what the issue is. This will be fixed in the next release, coming next week. Or you build/run from the current dev-branch.
It would be great if you could open a PR for your config and add it to the `assets/config` directory as well to the README in the supported devices table.

**Comment by tsterbak** [comment]
> You might edit the screen for selecting "restart bootloader" to include directions for the user to use the
> 
>     * volume keys to switch between options
> 
>     * power button to select
> 
> 
> Especially if this projects goal is to make it easy for _any non-technical person_ to install a different Android ROM.

You are right! The texts need some more work. Right now, I'm mostly working on making the experience better and easier from a technical perspective. But soon I will spend more time on texts.
Note, that these text about the bootloader is part of the config. So, if you want, you can propose a better version of the text for your device. This might be generalized at a later point in time.
